src/app/model/Knight.ts

```diff
@@ -0,0 +1,63 @@
+import { Board } from './Board';
+import { Piece } from './Piece';
+import { BOARDSIZE } from './Constances';
+import { PieceMovement } from './PieceMovement';
+import { columns, Row, Side, Square } from './Types';
+
+export class Knight extends Piece implements PieceMovement {
+  constructor(id: string, side: Side) {
+    super('uselessId', side);
+  }
+
+  possibleMoves(position: Square, board: Board): Square[] {
+    let movesToGo = [] as Square[];
+    movesToGo = movesToGo.concat(this.movesOnEmptyBoard(position, board));
+    // TODO: usuń pola zajęte
+    movesToGo = this.removeImpossibleSquares(movesToGo, board);
+    return movesToGo;
+  }
+
+  private movesOnEmptyBoard(position: Square, board: Board): Square[] {
+    const movesToGo: Square[] = [];
+    const currentColumnNumber = columns.indexOf(position.column);
+    const currentRowNumber = position.row;
+
+    if (this.isWithinChessboardBorders(currentColumnNumber + 2, currentRowNumber + 1)) {
```

Comment:
Te same uwagi co do Pawn - z wyciagnieciem do zmiennych itp :) Mozesz sprobowac to przerobic na podobne jak z lista ruchów i rekurencja jak zrobilismy w pionku.

Reply:
Obawiam się, że rekurencja w przypadku skoczka i króla nie ma racji bytu

Reply:
No może rzeczywiście, ale i tak za każdym razem mamy to if + obliczenia robisz po 2 razy to samo (dla ifa i dla psuha) no i mutujesz listę. Co byś powiedział na :) ? 


```
  private movesOnEmptyBoard(position: Square): Square[] {
    const currentColumnNumber = columns.indexOf(position.column);
    const currentRowNumber = position.row;

    return [
      { column: columns[currentColumnNumber + 2], row: (currentRowNumber + 1) as Row },
      { column: columns[currentColumnNumber + 2], row: (currentRowNumber - 1) as Row },
      { column: columns[currentColumnNumber + 1], row: (currentRowNumber + 2) as Row },
      { column: columns[currentColumnNumber + 1], row: (currentRowNumber - 2) as Row },
      { column: columns[currentColumnNumber - 1], row: (currentRowNumber + 2) as Row },
      { column: columns[currentColumnNumber - 1], row: (currentRowNumber - 2) as Row },
      { column: columns[currentColumnNumber - 2], row: (currentRowNumber + 1) as Row },
      { column: columns[currentColumnNumber - 2], row: (currentRowNumber - 1) as Row }
    ].filter(this.isWithinChessboardBorders)
  }

  private isWithinChessboardBorders(position: Square): boolean {
    const columnNumber = columns.indexOf(position.column);
    return columnNumber < BOARDSIZE && columnNumber >= 0 && position.row <= BOARDSIZE && position.row > 0;
  }
```

Reply:
Chyba jednak sie mylę

Reply:
jeśli da się te ify wywalić to jestem za :D będzie ładniej!